include/seqan3/search/algorithm/configuration/max_error.hpp

```diff
@@ -150,26 +209,19 @@ struct search_config_max_error_adaptor : public configuration_fn_base<search_con
      * \param[in] deletion_error The number of maximum deletion errors.
      * \returns A new configuration containing the max_error configuration element.
      */
-    template <typename configuration_t>
+    template <typename configuration_t, typename ... Args>
     //!\cond
         requires is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
     //!\endcond
     // TODO: replace int by templates and check whether it can be casted to uint8_t
-    // TODO: template argument packing. allow any order and subset of strong types
-    constexpr auto invoke(configuration_t && cfg, search_cfg::total<int> const total_error,
-                                                  search_cfg::substitution<int> const substitution_error,
-                                                  search_cfg::insertion<int> const insertion_error,
-                                                  search_cfg::deletion<int> const deletion_error) const
+    constexpr auto invoke(configuration_t && cfg, Args && ...args) const
     {
         static_assert(is_valid_search_configuration_v<search_cfg::id::max_error, remove_cvref_t<configuration_t>>,
                       SEQAN3_INVALID_CONFIG(search_cfg::id::max_error));
 
-        search_config_max_error tmp{std::tuple<uint8_t, uint8_t, uint8_t, uint8_t>{
-                                        static_cast<int>(total_error),
-                                        static_cast<int>(substitution_error),
-                                        static_cast<int>(insertion_error),
-                                        static_cast<int>(deletion_error)}
-                                    };
+        search_config_max_error tmp{{0, 0, 0, 0}};
+        set_values<false, false, false, false>(tmp, std::forward<Args>(args)...);
```

Comment:
i still think this could be done easier:
```cpp
template <typename in_t, typename out_t, typename t>
void get_type_out(in_t & in, out_t & out_t)
{
    static_assert(tuple_contains_type <= 1, "bla");
    if constexpr (type_list_contains<total> == 1)
        std::get<type>(out) = std::get<type>(in);
}

std::tuple<error, disjunction,...> out{0,0,0,0};
std::tuple in{args...};

foo<total>(in, out);
foo<disjunction>(in, out);
...
```